### PR TITLE
Additional PostgresTypes + tests

### PR DESCRIPTION
**Pull request by NGRichter**
Apparently I had my test environment not setup, all tests were ignored. I fixed that and now the PostgreSQL tests are working again.

A lot of types are aliases, found [here](https://www.postgresql.org/docs/current/datatype.html). 

Most types do not have parameters, some have 1 and numeric has 2. Copying a table should also correctly copy the datatype including parameters. Some types have parameters but can not be set as they are fixed. Examples are: 
smallint, integer and bigint with precision of 16, 32 and 64 and all with a scale of 0.
Date with a datetime precision of 0.

### CODE REVIEW

File: quantumdb-postgresql/src/main/java/io/quantumdb/core/schema/definitions/PostgresTypes.java [hunk 1]
```diff
@@ -12,39 +13,55 @@
...
 			case "boolean":
 				return PostgresTypes.bool();
 			case "varchar":
 			case "character varying":
-				return PostgresTypes.varchar(length);
+				return PostgresTypes.varchar();
+			case "char":
 			case "character":
-				return PostgresTypes.chars(length);
+				return PostgresTypes.chars(1);
```
Comment (NGRichter) [inline quantumdb-postgresql/src/main/java/io/quantumdb/core/schema/definitions/PostgresTypes.java:64]: The default parameter for the char or character data type is 1.

**Comment by NGRichter** [comment]
I can simplify the switch statement in PostgresTypes by using the default parameter value when it is not specified. It might increase the readability.

**Comment by michaeldejong** [comment]
I'll give this PR a review tomorrow.

File: quantumdb-postgresql/src/main/java/io/quantumdb/core/schema/definitions/PostgresTypes.java [hunk 2]
```diff
@@ -12,39 +13,55 @@
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public class PostgresTypes {
 
-	public static ColumnType from(String input) {
```
Comment (michaeldejong) [inline quantumdb-postgresql/src/main/java/io/quantumdb/core/schema/definitions/PostgresTypes.java:15]: It's a bit hard to follow from outside of this class which method you need now: `fromString` or `from`. I would suggest the following to improve on this:

- Rename `fromString` back to `from`.
- Make the current `from(String)` private, and rename it to something like `valueOf(String)` which is typically used in enums (even though this isn't an enum).

This leaves 3 public methods:
- `from(String)`, which can return all types.
- `from(String, int)` which can return types with a scalar.
- `from(String, int, int)` which can return types with a scalar and precision.

This means the caller can pass what ever info they have (a single string, or a string with additional scalar / precision values).

File: quantumdb-postgresql/src/main/java/io/quantumdb/core/schema/definitions/PostgresTypes.java [hunk 3]
```diff
@@ -12,39 +13,55 @@
...
 			String type = match.group(1);
-			int length = Integer.parseInt(match.group(2));
-			return from(type, length);
-		}
-		else {
-			return from(input, null);
+			int precision = Integer.parseInt(match.group(2));
+			int scale = Integer.parseInt(match.group(3));
+			return from(type.trim(), precision, scale);
+		} else {
```
Comment (michaeldejong) [inline quantumdb-postgresql/src/main/java/io/quantumdb/core/schema/definitions/PostgresTypes.java:23]: Bit of a nit-pick, but try to avoid bracket closing/opening on the same line. Bit of a styling thing TBF.

File: quantumdb-postgresql/src/main/java/io/quantumdb/core/schema/definitions/PostgresTypes.java [hunk 4]
```diff
@@ -54,15 +71,77 @@ public static ColumnType from(String type, Integer length) {
...
+			case "decimal":
+			case "numeric":
+				return PostgresTypes.numeric(length);
+			case "varchar":
+			case "character varying":
+				return PostgresTypes.varchar(length);
+			case "char":
+			case "character":
+				return PostgresTypes.chars(length);
+			case "timestamp with time zone":
```
Comment (michaeldejong) [inline quantumdb-postgresql/src/main/java/io/quantumdb/core/schema/definitions/PostgresTypes.java:122]: There's an alias for this as well that PG supports: `timestamptz`.

File: quantumdb-postgresql/src/test/java/io/quantumdb/core/backends/integration/videostores/AddColumnToCustomersTable.java [hunk 5]
```diff
@@ -103,7 +105,11 @@ public void verifyTableStructure() {
 				.addColumn(new Column("customer_id", integer(), NOT_NULL))
 				.addColumn(new Column("rental_id", integer(), NOT_NULL))
 				.addColumn(new Column("date", date(), NOT_NULL))
-				.addColumn(new Column("amount", floats(), NOT_NULL));
+				.addColumn(new Column("amount", floats(), NOT_NULL))
+				.addColumn(new Column("amount_numeric", numeric(), NOT_NULL))
+				.addColumn(new Column("amount_numeric_precision", numeric(10), NOT_NULL))
+				.addColumn(new Column("amount_numeric_precision_scale", numeric(10,2), NOT_NULL))
+				.addColumn(new Column("invoice_pdf", bytea(), NOT_NULL));
```
Comment (michaeldejong) [inline quantumdb-postgresql/src/test/java/io/quantumdb/core/backends/integration/videostores/AddColumnToCustomersTable.java:112]: Maybe it would be best to make a `PostgresTypesTest` class instead to verify all of these can be parsed correctly. The videostores integration test suite was meant for debugging some more complex migrations (and avoid future regressions). They weren't really meant to test database types.

File: quantumdb-postgresql/src/test/java/io/quantumdb/core/backends/integration/videostores/MakeStoreFieldInStaffTableNullable.java [hunk 6]
```diff
@@ -24,10 +26,9 @@
 import org.junit.Assert;
 import org.junit.BeforeClass;
 import org.junit.ClassRule;
-import org.junit.Ignore;
 import org.junit.Test;
 
-@Ignore // TODO: There seems to be something wrong with this test. Must investigate further.
+// @Ignore - Seems to be working again
```
Comment (michaeldejong) [inline quantumdb-postgresql/src/test/java/io/quantumdb/core/backends/integration/videostores/MakeStoreFieldInStaffTableNullable.java:31]: If so, feel free to drop this line and reenable the test. Thanks for checking 👍

**Review by michaeldejong — CHANGES_REQUESTED** [review]
Aside from the comments on `PostgresTypes`, I think this could benefit from having its own test suite instead of extending the videostores example.

But aside from that, this is very impressive. Nice work!

File: quantumdb-postgresql/src/test/java/io/quantumdb/core/backends/integration/videostores/MakeStoreFieldInStaffTableNullable.java (hunk 6 above)
Comment (NGRichter) [reply quantumdb-postgresql/src/test/java/io/quantumdb/core/backends/integration/videostores/MakeStoreFieldInStaffTableNullable.java]: It worked when I added my extra columns, right now it breaks again... @Ignore once again

File: quantumdb-postgresql/src/main/java/io/quantumdb/core/schema/definitions/PostgresTypes.java [hunk 7]
```diff
@@ -13,78 +14,145 @@
...
+				error += " (1st argument: " + precision + ", 2nd argument: " + scale + ")";
 				throw new IllegalArgumentException(error);
 		}
 	}
 
 	public static ColumnType oid() {
-		return new ColumnType(ColumnType.Type.OID, false, "oid", () -> 0L,
-				(statement, position, value) -> statement.setLong(position, ((Number) value).longValue()));
+		return new ColumnType(ColumnType.Type.OID, false, "oid", () -> 0,
+				(statement, position, value) -> statement.setInt(position, ((Number) value).intValue()));
```
Comment (NGRichter) [inline quantumdb-postgresql/src/main/java/io/quantumdb/core/schema/definitions/PostgresTypes.java:155]: According to the Postgresql documentation oid is an unsigned 4 byte integer

File: quantumdb-postgresql/src/main/java/io/quantumdb/core/schema/definitions/PostgresTypes.java [hunk 8]
```diff
@@ -106,56 +179,87 @@ public static ColumnType text() {
...
 				(statement, position, value) -> statement.setDate(position, (java.sql.Date) value));
 	}
 
-	public static ColumnType doubles() {
-		return new ColumnType(ColumnType.Type.DOUBLE, false, "real", () -> 0.00d,
-				(statement, position, value) -> statement.setDouble(position, (Double) value));
-	}
-
-	public static ColumnType floats() {
-		return new ColumnType(ColumnType.Type.FLOAT, false, "double precision", () -> 0.00f,
```
Comment (NGRichter) [inline quantumdb-postgresql/src/main/java/io/quantumdb/core/schema/definitions/PostgresTypes.java:152]: These were switched around, "real" is a float and "double precision" is a double

**Comment by NGRichter** [comment]
According to the tests I wrote, everything works now. 
I tried to have tables with all possible types and checked if what I put in also came out.
Additionally I also copied the tables and tested again if the values I put in also were retrieved as expected.

**Review by michaeldejong — APPROVED** [review]
